engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetStateParameter.java

```diff
@@ -48,37 +50,24 @@ public int hashCode() {
 
     @Override
     public boolean equals(Object obj) {
-        return (obj instanceof SetStateParameter) && this.enabled == ((SetStateParameter) obj).isEnabled() && this.glParameter == ((SetStateParameter) obj).glParameter;
+        return (obj instanceof SetStateParameter) && this.enabled == ((SetStateParameter) obj).enabled && this.glParameter == ((SetStateParameter) obj).glParameter;
     }
 
-    public boolean isEnabled() {
-        return enabled;
+    private String getStatus() {
+        return enabled? "Enabled": "Disabled";
     }
 
     @Override
-    public RenderPipelineTask generateTask() {
-        if (enabled) {
-            return getEnablingTask();
-        } else {
-            return getDisablingTask();
-        }
+    public String toString() {
+        return String.format("%30s: %s", this.getClass().getSimpleName(), getStatus());
     }
 
-    protected abstract RenderPipelineTask getDisablingTask();
-
-    protected abstract RenderPipelineTask getEnablingTask();
-
-    public String getStatus() {
-        String status = "disabled";
+    @Override
+    public void process() {
         if (enabled) {
-            status = "enabled";
+            glEnable(glParameter);
+        } else {
+            glDisable(glParameter);
```

Comment:
Similar to previous comments this class now needs to be split into an `EnableStateParameter `and `DisableStateParameter`. As the defaults vary from parameter to parameter you'll probably have to extend a common class which could be the existing `SetStateParameter`.